### ISSUE TITLE
[3.3.5] Polymorph & Grounding Totem

### ISSUE THREAD

**Issue by Persephone** [TrinityCore/TrinityCore#21459]
**Description:**

Casting polymorph on a shaman redirects it properly to the grounding totem, but the grounding totem turns into an actual sheep and starts walking around instead of just "eating it" and then dissapearing.

**Current behaviour:**

Grounding Totem turns into a sheep.

**Expected behaviour:**

Should just eat the polymorph and dissapear.

**Steps to reproduce the problem:**

1. Cast grounding totem as shaman.
2. Cast polymorph on shaman.

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

https://github.com/TrinityCore/TrinityCore/commit/1cc3d33cfe7e51d66925d5df536a952165a88094

**TDB version:** 

TDB 335.64

**Operating system:** Ubuntu

**Comment by Keader** [comment]
cool bug xD

**Comment by Persephone** [comment]
Ya xD looks so odd

**Comment by tkrokli** [comment]
Looks like this happens because the totem is a de facto NPC and the grounding totem redirects the spell onto itself, while the check for humanoid target is skipped.

**Comment by Persephone** [comment]
@tkrokli  What do you mean about it being a de facto npc? Do you know whats causing it?

**Comment by kvipka** [comment]
@Persephone , look at this : 
https://dbwotlk.com/?spell=8177 - spell summoning totem (npc Grounding Totem https://dbwotlk.com/?npc=5925)

**Comment by tkrokli** [comment]
spell Grounding Totem: https://woehead.way-of-elendil.fr/?spell=8177
`Effect #1`  Summon: ([Grounding Totem](https://woehead.way-of-elendil.fr/?npc=5925)) https://woehead.way-of-elendil.fr/?npc=5925
- http://www.wowhead.com/npc=5925/grounding-totem

Grounding Totem is NPC entry 5925 (creature)

**Comment by Persephone** [comment]
Ya its an npc summoned, but what does that have to do with the humanoid check?

**Comment by tkrokli** [comment]
Maybe I don't know the spell well enough, but can you polymorph anything into a sheep in-game? (Beast, Dragonkin, Demon, Elemental, Giant, Undead, Humanoid, Critter, Mechanical, Totem, Non-Combat Pet, Gas Cloud, Wild Pet)

**Comment by kvipka** [comment]
beast 100% available, mechanical no.

**Comment by tkrokli** [comment]
Never mind what causes the visual. I presume this will be dealt with when we are tired of laughing at the Grounding Sheep 🐑 - I mean Totem :-P

**Comment by Persephone** [comment]
Haha I am tired of seeing it xD wish i knew how to fix it

**Comment by tkrokli** [comment]
Perhaps you could check SpellWork for how the related spells are handled:
https://woehead.way-of-elendil.fr/?spell=8178 (Grounding Totem Effect)
Apply Area Aura Party: Spell Magnet
https://woehead.way-of-elendil.fr/?spell=8179 (Grounding Totem Passive)
Apply Aura: Periodic Trigger Spell

Also check if the Polymorph Visual used by the totem is the same as
https://woehead.way-of-elendil.fr/?spell=32826 (Polymorph Cast Visual)

**Comment by kvipka** [comment]
don't touch grounding totem!

**Comment by tkrokli** [comment]
OK, so you believe this is caused by something very distant from the totem itself?
Feel free to elaborate, if you have got any ideas.

**Comment by kvipka** [comment]
Ok, i did temp solution for you. I think @ariel-  will rewrite it, but right now you can use this a little present from rustemu.

video: https://www.youtube.com/watch?v=Fa0R-coPgTQ&feature=youtu.be

**Comment by Persephone** [comment]
yeah that works @kvipka :) hope ariel can use it to fix it properly, but thanks for sharing!! <3 <3

**Comment by ariel-** [comment]
Current magnet implementation system is a hack

**Comment by EricksOliveira** [comment]
Does anyone have a prediction to fix this problem?

**Comment by illfated** [comment]
That is odd. I thought this was solved already, but I must have confused this issue with a different totem issue.

**Comment by EricksOliveira** [comment]
I used the temporary change of @kvipka , actually he did right with Polymorph, but with Fear no. The totem receives the fear, but remains active.

**Comment by EricksOliveira** [comment]
That happens. He runs on Fear and does not disappear. You can cast Fear and the totem running. haha More only with the @kvipka !

**Comment by kvipka** [comment]
heh, okay guys, will implement attr and for fear too, when will home. But this issue was about only confuse )

**Comment by Eliminationzx** [comment]
Nice hackfix.
But SPELL_AURA_MAGNET should work through proc system.
If magnet aura proc then -> kill totem.

**Comment by offl** [comment]
Still valid in aba07047a05f
